docs/node/run-your-node/prerequisites/system-configuration.mdx

```diff
@@ -127,24 +69,83 @@ Add the following task to your playbook:
 </TabItem>
 </Tabs>
 
-:::tip
+:::
+
+## Increase File Descriptor Limit
+
+Make sure that the user under which you are running your Oasis Node has a
+high-enough file descriptor limit as the databases can have many files open and
+running out of file descriptors can lead to the node stopping unexpectedly.
+
+You can check the file descriptor limit by running the following as the same
+user that will run Oasis Node:
+
+```bash
+ulimit -n
+```
+
+If this number is lower than 102400 you should consider increasing it by
+updating your system configuration. You can configure _temporary_ limits by
+running:
+
+```bash
+ulimit -n 102400
+```
+
+Note that this limit only applies to any processes started from the same shell
+after the command was executed. If you want to make the change _permanent_, you
+have the following options:
+
+<Tabs>
+<TabItem value="System config file">
+
+As `root`, create a System-wide Resource Limits Configuration File in
+`/etc/security/limits.d/99-oasis-node.conf` with content similar to the
+following example:
+
+```bash
+*        soft    nofile    102400
+*        hard    nofile    1048576
+```
+
+You can replace `*` with the name of the user that is running the Oasis Node in
+case you only want to change the limits for that particular user.
+
+:::caution
 
-Setting `oasis` user's Shell to `/usr/sbin/nologin` prevents (accidentally)
-logging-in as this user.
+In order for the changes to take effect a system restart may be required.
 
 :::
 
-2. Stop your Oasis Node.
+</TabItem>
+<TabItem value="Systemd">
 
-3. Transfer ownership of the datadir to the `oasis` user:
+In case you are running your Oasis Node process via
+[systemd](https://systemd.io/), you can add the following directive under the
+`[Service]` section:
 
-```sh
-chown -R oasis /node/data
+```text
+LimitNOFILE=102400
+```
+
+</TabItem>
+<TabItem value="Docker">
+
+If you are running Oasis Node via [Docker](https://www.docker.com/) you can pass
+the following option to `docker run` in order to increase the limit to desired
+values:
+
+```text
+--ulimit nofile=102400:1048576
 ```
 
-See [Invalid Permissions] troubleshooting guide for more information.
+</TabItem>
+</Tabs>
 
-4. Update how you run Oasis Node:
+## Example snippets for different setups
+
+You may find the following snippets helpful in case you are running `oasis-node`
+process with `systemd`, `Docker` or `runit`.
```

Comment:
I would avoid ticks here since they are not commands.

```suggestion
You may find the following snippets helpful in case you are running `oasis-node`
process with systemd, Docker or runit.
```